app/models/http_proxy.rb

```diff
@@ -0,0 +1,25 @@
+class HttpProxy < ActiveRecord::Base
+  include Authorizable
+  include Taxonomix
+  include Encryptable
+
+  extend FriendlyId
+
+  encrypts :password
+  friendly_id :name
+
+  has_many :compute_resources
+
+  validates :url, :presence => true
+  validates :port, :presence => true
+
+  scoped_search :on => :name
```

Comment:
I could see being allowed to search on URL as well in the case where you remember the proxy domain name but forgot what you named the proxy.